### PR TITLE
Create SLD to agency name and CyHy ID mapping collection

### PR DESCRIPTION
**Pull request by jsf9k**
This data is already compiled and used in-memory in the BOD 18-01 HTTPS and Trustworthy Email reporting.  The CT work that @felddy is doing in [dhs-ncats/cyhy-ct-logs](https://github.com/dhs-ncats/cyhy-ct-logs) requires this data as well, so we will save it in the `scan` database.

### CODE REVIEW

File: create_sld_to_agency_name_and_id_mapping.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+DB_CONFIG_FILE = '/run/secrets/scan_write_creds.yml'
+INCLUDE_DATA_DIR = '/home/saver/include/'
+SHARED_DATA_DIR = '/home/saver/shared/'
+
+AGENCIES_FILE = INCLUDE_DATA_DIR + 'agencies.csv'
+CURRENT_FEDERAL_FILE = SHARED_DATA_DIR + 'artifacts/current-federal_modified.csv'
+
+
+def db_from_config(config_filename):
```
Comment (dav3r) [inline create_sld_to_agency_name_and_id_mapping.py:16]: This doesn't need to happen today, but at some point we should import this function from the CyHy library.

File: create_sld_to_agency_name_and_id_mapping.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    #
+    # It would be great to check the return code here, but I don't see
+    # anything of use in DeleteResult:
+    # http://api.mongodb.com/python/current/api/pymongo/results.html#pymongo.results.DeleteResult
+    db.domains.delete_many({})
+
+    # Now add our new results
+    res = db.domains.insert_many(records)
+    if len(res.inserted_ids) != len(records):
+        print(f'Unable to write new SLD records from "{db.name}" '
```
Comment (dav3r) [inline create_sld_to_agency_name_and_id_mapping.py]: I'd like to see this error message also mention how many records were actually written to the database vs. how many were supposed to be written.

File: create_sld_to_agency_name_and_id_mapping.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    # Drop all previous documents from domains collection
+    #
+    # It would be great to check the return code here, but I don't see
+    # anything of use in DeleteResult:
+    # http://api.mongodb.com/python/current/api/pymongo/results.html#pymongo.results.DeleteResult
+    db.domains.delete_many({})
+
+    # Now add our new results
+    res = db.domains.insert_many(records)
```
Comment (dav3r) [inline create_sld_to_agency_name_and_id_mapping.py]: As discussed, we would like to see the insert/update process look like this:
* Mark all domain documents as `latest:False`
* [Update with upsert](https://docs.mongodb.com/manual/reference/method/db.collection.updateMany/) all new domain documents with `latest:True`

File: create_sld_to_agency_name_and_id_mapping.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    # Set up the scan database connection
+    db = db_from_config(DB_CONFIG_FILE)
+
+    # Drop all previous documents from domains collection
+    #
+    # It would be great to check the return code here, but I don't see
+    # anything of use in DeleteResult:
+    # http://api.mongodb.com/python/current/api/pymongo/results.html#pymongo.results.DeleteResult
+    db.domains.delete_many({})
```
Comment (dav3r) [inline create_sld_to_agency_name_and_id_mapping.py]: This step will no longer be needed once we use update/upsert as described in the next comment.

**Review by dav3r — APPROVED** [review]
Approved, but please see requested changes and implement at an appropriate time

File: create_sld_to_agency_name_and_id_mapping.py (hunk 1 above)
Comment (jsf9k) [reply create_sld_to_agency_name_and_id_mapping.py:16]: :+1:

**Comment by h-m-f-t** [comment]
This is so great 🇺🇸

File: create_sld_to_agency_name_and_id_mapping.py (hunk 3 above)
Comment (jsf9k) [reply create_sld_to_agency_name_and_id_mapping.py]: We ended up doing this using the `scan_date` flag.  That way we don't need to add a `latest` boolean.

**Comment by jsf9k** [comment]
@dav3r, do you want to take another look since we changed a lot of lines?  If you're still good I'll merge.

I did test this in my `jsf9k` workspace.

**Review by dav3r — APPROVED** [review]
Looking solid now @jsf9k - I agree that transactions are the way to go here. 🏎

**Pull request merged**